homeassistant/components/roborock/image.py

```diff
@@ -74,50 +71,24 @@ async def async_added_to_hass(self) -> None:
         """When entity is added to hass load any previously cached maps from disk."""
         await super().async_added_to_hass()
         content = await self.coordinator.map_storage.async_load_map(self.map_flag)
-        self.cached_map = content or b""
+        if self.coordinator.maps[self.map_flag].image is None:
+            self.coordinator.maps[self.map_flag].image = content
         self._attr_image_last_updated = dt_util.utcnow()
```

Comment:
Should this be consistent with `_handle_coordinator_update` and use the same methodology for setting the updated timestamp?